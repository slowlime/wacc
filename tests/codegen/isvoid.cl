class Main {
    int : Int;
    string : String;
    bool : Bool;
    object : Object;

    io : IO <- new IO;

    main(): Object {{
        print_is_void(int);
        print_is_void(string);
        print_is_void(bool);
        print_is_void(object);

        print_is_void(while false loop 0 pool);
    }};

    print_is_void(x : Object): Object {
        if isvoid x then
            io.out_string("null\n")
        else
            io.out_string("not null\n")
        fi
    };
};
