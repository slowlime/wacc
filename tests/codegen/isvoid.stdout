not null
not null
not null
null
null
